src/main/perl/FileEditor.pm

```diff
@@ -58,13 +66,45 @@ sub _is_valid_source
     return -f $fn;
 }
 
+sub _is_reference_newer
```

Comment:
This sub routine lacks any POD. It also hardcodes references to the internal data structures which makes it hard to follow: I have to know what *$self->{options}->{source} and *$self->{filename} refer to in order to grok it. Could it be made more general, so that the things/references to be compared are passed as arguments? Then the function could conceivably be reused in other places too.